utils/useAuth.js

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react';
+import { supabase } from './initSupabase';
+
+const postData = (url, data = {}) =>
+  fetch(url, {
+    method: 'POST',
+    headers: new Headers({ 'Content-Type': 'application/json' }),
+    credentials: 'same-origin',
+    body: JSON.stringify(data),
+  }).then((res) => res.json());
+
+const useAuth = () => {
+  const [session, setSession] = useState(null);
+  const [user, setUser] = useState(null);
```

Comment:
Since `useEffect` runs async after the first render, the first render won't include session data (which could result in a flash as the session immediately updates). Maybe we could set the initial state to the session?

This would be an awesome helper to have available in the JS library, otherwise I'd be re-writing it for every app. (Or maybe a new react helpers library :P)

Reply:
We created some cookie helpers and some React auth components. Please refer to this example: https://github.com/supabase/supabase/tree/master/examples/nextjs-with-supabase-auth